libcloudforensics/aws.py

```diff
@@ -0,0 +1,695 @@
+        break
+
+    return instances
+
+  def ListVolumes(self, region=None, filters=None):
+    """List volumes of an AWS account.
+
+    Args:
+      region (str): Optional. The region from which to list the volumes.
+      filters (list(dict)): Optional. Filter for the query.
+
+    Returns:
+      dict: Dictionary with name and metadata for each volume.
+
+    Raises:
+      RuntimeError: If volumes can't be listed.
+
+    Example usage:
+      # List volumes attached to the instance 'some-instance-id'
+      ListVolumes(filters=[
+          dict(Name='attachment.instance-id', Values=['some-instance-id'])])
+    """
+    if not filters:
+      filters = []
+
+    volumes = {}
+    next_token = None
+    client = self.ClientApi(EC2_SERVICE, region=region)
+
+    while True:
+      try:
+        if next_token:
+          response = client.describe_volumes(
+              Filters=filters, NextToken=next_token)
+        else:
+          response = client.describe_volumes(Filters=filters)
+      except client.exceptions.ClientError as exception:
+        raise RuntimeError('Error: could not retrieve volumes: {0:s}'.format(
+            str(exception)))
+
+      for volume in response['Volumes']:
+        volume_info = {
+            'region': self.default_region,
+            'zone': volume['AvailabilityZone']
+        }
+
+        for tag in volume.get('Tags', []):
+          if tag.get('Key') == 'Name':
+            volume_info['name'] = tag.get('Value')
+            break
+
+        if len(volume['Attachments']) > 0:
+          volume_info['device'] = volume['Attachments'][0]['Device']
+
+        volumes[volume['VolumeId']] = volume_info
+
+      next_token = response.get('NextToken')
+      if not next_token:
+        break
+
+    return volumes
+
+  def GetInstancesByNameOrId(self, instance_name=None,
+                             instance_id=None, region=None):
+    """Get instances from an AWS account by their name tag or an ID.
+
+    Exactly one of [instance_name, instance_id] must be specified. If looking up
+    an instance by its ID, the method returns a list with exactly one
+    element. If looking up instances by their name tag (which are not unique
+    across instances), then the method will return a list of all instances
+    with that name tag, or an empty list if no instances with matching name
+    tag could be found.
+
+    Args:
+      instance_name (str): Optional. The instance name tag of the instance to
+          get.
+      instance_id (str): Optional. The instance id of the instance to get.
+      region (str): Optional. The region to look the instance in.
+
+    Returns:
+      list(AWSInstance): A list of Amazon EC2 Instance objects.
+
+    Raises:
+      ValueError: If both instance_name and instance_id are None or if both
+          are set.
+    """
+    if not instance_name and not instance_id or instance_name and instance_id:
+      raise ValueError('Error: you must specify exactly one of [instance_name, '
```

Comment:
Probably can remove 'Error: ' in the string message of a ValueError exception =)

Reply:
Same thing at other lines